models/gif_cond_exp.h

```diff
@@ -0,0 +1,450 @@
+   * Hiding
+   */
+  using Node::handle;
+  using Node::handles_test_event;
+
+  port send_test_event( Node&, rport, synindex, bool );
+
+  void handle( SpikeEvent& );
+  void handle( CurrentEvent& );
+  void handle( DataLoggingRequest& );
+
+  port handles_test_event( SpikeEvent&, rport );
+  port handles_test_event( CurrentEvent&, rport );
+  port handles_test_event( DataLoggingRequest&, rport );
+
+
+  void get_status( DictionaryDatum& ) const;
+  void set_status( const DictionaryDatum& );
+
+private:
+  void init_state_( const Node& proto );
+  void init_buffers_();
+  void calibrate();
+
+  void update( Time const&, const long_t, const long_t );
+
+  // make dynamics function quasi-member
+  friend int gif_cond_exp_dynamics( double, const double*, double*, void* );
+
+  // The next two classes need to be friends to access the State_ class/member
+  friend class RecordablesMap< gif_cond_exp >;
+  friend class UniversalDataLogger< gif_cond_exp >;
+
+  // ----------------------------------------------------------------
+
+  /**
+   * Independent parameters of the model.
+   */
+  struct Parameters_
+  {
+    double_t g_L_;
+    double_t E_L_;
+    double_t V_reset_;
+    double_t Delta_V_;
+    double_t V_T_star_;
+    double_t lambda_0_; /** 1/ms */
+
+    double_t E_ex_; /** Excitatory reversal Potential in mV */
+    double_t E_in_; /** Inhibitory reversal Potential in mV  */
+    double_t
+      tau_synE_; /** Synaptic Time Constant for Excitatory Synapse in ms */
+    double_t
+      tau_synI_; /** Synaptic Time Constant for Inhibitory Synapse in ms */
+
+    /** Refractory period in ms. */
+    double_t t_ref_;
+
+    /** Membrane capacitance in pF. */
+    double_t c_m_;
+
+    /** We use stc and sfa, respectively instead of eta and gamma
+    (mentioned in the references). */
+
+    /** List of spike-triggered current time constant in ms. */
+    std::vector< double_t > tau_stc_;
+
+    /** List of spike-triggered current jumps in nA. */
+    std::vector< double_t > q_stc_;
+
+    /** List of adaptive threshold time constant in ms. */
+    std::vector< double_t > tau_sfa_;
+
+    /** List of adaptive threshold jumps in mV. */
+    std::vector< double_t > q_sfa_;
+
+    /** External DC current. */
+    double_t I_e_;
+
+    Parameters_(); //!< Sets default parameter values
+
+    void get( DictionaryDatum& ) const; //!< Store current values in dictionary
+    void set( const DictionaryDatum& ); //!< Set values from dictionary
+  };
+
+  // ----------------------------------------------------------------
+
+  /**
+   * State variables of the model.
+   */
+  struct State_
+  {
+    //! Symbolic indices to the elements of the state vector y
+    enum StateVecElems
+    {
+      V_M = 0,
+      G_EXC,
+      G_INH,
+      STATE_VEC_SIZE
+    };
+
+    double_t
+      y_[ STATE_VEC_SIZE ]; //!< neuron state, must be C-array for GSL solver
+
+    double_t y0_;  //!< This is piecewise constant external current
```

Comment:
We are trying to get rid of the old ``y0_``, ``y1_``... names in favor of more explicit ones; could you adapt this?